src/java.base/share/classes/java/util/jar/JarOutputStream.java

```diff
@@ -78,8 +78,12 @@ public JarOutputStream(OutputStream out) throws IOException {
      * to the start of the entry data. This method will also close
      * any previous entry. The default compression method will be
      * used if no compression method was specified for the entry.
-     * The current time will be used if the entry has no set modification
-     * time.
+     * When writing a compressed (i.e. {@code ZipEntry.DEFLATED}))
+     * entry, and the compressed size has not been explicitly set with
+     * the {@link ZipEntry#setCompressedSize(long)} method, then the
+     * compressed size will be set to the actual compressed size after
+     * deflation. The current time will be used if the entry has no set
+     * modification time.
```

Comment:
Thanks for combining the wording, I think this looks good. I'd probably drop "i.e." so that it's just "DEFLATED" parentheses.

Reply:
I might consider making "The current time..." its own paragraph separate from the compression discussion as I think it would be clearer that way.